test/user_test.rb

```diff
@@ -0,0 +1,56 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require './lib/user'
+
+class UserTest < Minitest::Test
+
+  def test_it_exists
+    sal = User.new("Sal")
+    assert_equal "Sal", sal.name
+    assert_equal [], sal.jokes
+  end
+
+  def test_it_has_jokes
+    joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+    assert_equal 1, joke_1.joke
+    assert_equal 2, joke_2.joke
+  end
+
+  def test_can_learn_joke
+    sal = User.new("Sal")
+    joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+    sal.learn(joke_1)
+    sal.learn(joke_2)
+    assert_instance_of Joke, sal.jokes[0]
```

Comment:
A better strategy here is asserting that sal now holds the same jokes that we instantiated in our test, in our `jokes` array that lives on the sal `User` object. This makes our test more powerful.

Something like: 
`assert_equal [joke_1, joke_2], sal.jokes`